plugin/scalpel.vim

```diff
@@ -337,13 +344,21 @@ execute 'vnoremap <Plug>(ScalpelVisual) :' .
       \ s:command .
       \ "/\\v<<C-R>=escape(scalpel#cword(<SID>GetCurpos()), " . g:magic_chars . ")<CR>>//<Left>"
 
+" Change all instances of selection
+execute 'vnoremap <Plug>(ScalpelVisualSelection) :<C-u>' .
+      \ s:command .
+      \ "/\\v<C-R>=escape(scalpel#get_visual_selection(), " . g:magic_chars . ")<CR>//<Left>"
+execute 'vnoremap <Plug>(ScalpelVisualSmart) :<C-u>' .
+    \ s:command .
+    \ "/\\v<C-R>=escape(<SID>TransformSelection(scalpel#get_visual_selection()), " . g:magic_chars . ")<CR>//<Left>"
+
 let s:map=get(g:, 'ScalpelMap', 1)
 if s:map
   if !hasmapto('<Plug>(Scalpel)') && maparg('<leader>e', 'n') ==# ''
     nmap <unique> <Leader>e <Plug>(Scalpel)
   endif
-  if !hasmapto('<Plug>(ScalpelVisual)') && maparg('<leader>e', 'v') ==# ''
-    vmap <unique> <Leader>e <Plug>(ScalpelVisual)
+  if !hasmapto('<Plug>(ScalpelVisualSmart)') && maparg('<leader>e', 'v') ==# ''
+    vmap <unique> <Leader>e <Plug>(ScalpelVisualSmart)
```

Comment:
Because of the way people tend to install Vim plugins (ie. with a plug-in manager, pointing at `HEAD` of `master`), we can't really _ever_ change behavior in breaking ways. And that includes breaking it when we bump the major version number, because people never see the version number, and therefore never have the opportunity to decline/avoid the breakage.

So that means it is OK to _add_ functionality (like the new `ScalpelVisualSmart` and ScalpelVisualSelection`), but not OK to overwrite previously existing functionality with it.

That is, you can pick a different mapping for it, or not map it at all — and instead mention it in the docs. People who are interested are then free to assign it to `<leader>e` if they want.